src/Service/VendorService/OverDrive/OverDriveMagazinesVendorService.php

```diff
@@ -0,0 +1,221 @@
+use App\Service\VendorService\OverDrive\Api\Client;
+use App\Service\VendorService\ProgressBarTrait;
+use App\Utils\Message\VendorImportResultMessage;
+use App\Utils\Types\IdentifierType;
+use Doctrine\ORM\EntityManagerInterface;
+use GuzzleHttp\ClientInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Class OverDriveMagazinesVendorService.
+ */
+class OverDriveMagazinesVendorService extends AbstractBaseVendorService
+{
+    use ProgressBarTrait;
+
+    protected const VENDOR_ID = 16;
+
+    private const VENDOR_SEARCH_TERM = 'facet.acSource="ereolen magazines"';
+    private const VENDOR_MAGAZINE_URL_BASE = 'http://link.overdrive.com/';
+
+    private $searchService;
+    private $httpClient;
+    private $apiClient;
+
+    /**
+     * OverDriveMagazinesVendorService constructor.
+     *
+     * @param eventDispatcherInterface $eventDispatcher
+     *   Dispatcher to trigger async jobs on import
+     * @param entityManagerInterface $entityManager
+     *   Doctrine entity manager
+     * @param loggerInterface $statsLogger
+     *   Logger object to send stats to ES
+     * @param SearchService $searchService
+     *   Datawell search service
+     * @param ClientInterface $httpClient
+     *   Http client to send api requests
+     * @param Client $apiClient
+     *   Api client for the OverDrive API
+     */
+    public function __construct(EventDispatcherInterface $eventDispatcher, EntityManagerInterface $entityManager, LoggerInterface $statsLogger, SearchService $searchService, ClientInterface $httpClient, Client $apiClient)
+    {
+        parent::__construct($eventDispatcher, $entityManager, $statsLogger);
+
+        $this->searchService = $searchService;
+        $this->httpClient = $httpClient;
+        $this->apiClient = $apiClient;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function load(): VendorImportResultMessage
+    {
+        if (!$this->acquireLock()) {
+            return VendorImportResultMessage::error(parent::ERROR_RUNNING);
+        }
+
+        $this->loadConfig();
+
+        $this->progressStart('Search data well for: "'.self::VENDOR_SEARCH_TERM.'"');
+
+        $offset = 1;
+        try {
+            do {
+                $this->progressMessage('Search data well for: "'.self::VENDOR_SEARCH_TERM.'" (Offset: '.$offset.')');
+
+                // Search the data well for material with acSource set to "ereolen magazines".
+                [$pidArray, $more, $offset] = $this->searchService->search(self::VENDOR_SEARCH_TERM, $offset);
+
+                // Get the OverDrive APIs title urls from the results
+                $pidArray = $this->getTitleUrls($pidArray);
+
+                // Get the OverDrive APIs crossRefIds from the results
+                $pidArray = $this->getTitleIds($pidArray);
+
+                // Get the OverDrive cover urls
+                $pidArray = $this->getCoverUrls($pidArray);
```

Comment:
So all these call's changes the pid's form  the data well og filters something out of the array?